lib/spack/spack/build_systems/python.py

```diff
@@ -102,6 +94,49 @@ def configure(self, spec, prefix):
 
     py_namespace = None
 
+    @property
+    def import_modules(self):
+        """Name of modules that the Python package provides.
+
+        During test time, Spack tries to import these modules to determine
+        whether or not the installation succeeded.
+
+        Returns:
+            list: list of Python modules provided by the package
+        """
+        # NOTE: In the Python ecosystem, a distinction is made between
+        # 'packages' and 'modules'. 'packages' are directories containing
+        # `__init__.py` files, while 'modules' are standalone `*.py` files.
+
+        # First, we try using `setuptools.find_packages` to get a list of
+        # 'packages' provided by this Spack package.
+        packages = setuptools.find_packages(os.path.join(
+            inspect.getmodule(self).prefix,
+            inspect.getmodule(self).site_packages_dir))
+
+        if packages:
+            return packages
+
+        # If the package only contains Python 'modules', we guess using
+        # the name of the package.
+        module = self.name
+
+        # Spack often prepends `py-` to the name of the package.
+        if module.startswith('py-'):
+            module = module[3:]
+
+        # Many libraries with Python bindings add `python-` to the beginning
+        # or `-python` to the end of the name of the package.
+        if module.startswith('python-'):
+            module = module[7:]
+        if module.endswith('-python'):
+            module = module[:-7]
+
+        # Python imports either use foo.bar or foo_bar, never foo-bar.
+        module = module.replace('-', '.')
```

Comment:
Both `foo.bar` and `foo_bar` are uncommon. I think `foo.bar` is slightly more common, so I chose this one. If we ever implement #2281/#2718, we will no longer need to guess, as each package will have a `pypi` attribute with the package name.